ssl/s3_lib.c

```diff
@@ -3674,7 +3674,7 @@ long ssl3_ctrl(SSL *s, int cmd, long larg, void *parg)
         {
             unsigned int id;
 
-            if (SSL_CONNECTION_IS_TLS13(sc) && sc->s3.did_kex)
+            if ((SSL_CONNECTION_IS_TLS13(sc) || SSL_CONNECTION_IS_DTLS13(sc)) && sc->s3.did_kex)
```

Comment:
We really need a `SSL_CONNECTION_IS_13` or similar....I thought you'd already added something like that in one of the previous PRs, but I can't now find it.

Reply:
I agree. It is part of this:
https://github.com/openssl/openssl/pull/22366

I'll rebase this PR and update once https://github.com/openssl/openssl/pull/22366 is merged.